### PR TITLE
MuOxi on Docker

### PR DESCRIPTION
**Pull request by n0um3n4**
#### What does this PR do?

- Add Docker configuration files to run MuOxi on docker.

With Docker

### CODE REVIEW

File: muoxi/src/staging/prelude.rs [hunk 1]
```diff
@@ -40,10 +40,10 @@ pub type LinesCodecResult<T> = Result<T, LinesCodecError>;
 pub type CommandResult<T> = Result<T, &'static str>;
 
 /// Current listening port of the MuOxi game engine
-pub static GAME_ADDR: &'static str = "127.0.0.1:4567";
+pub static GAME_ADDR: &'static str = "0.0.0.0:4567";
```
Comment (n0um3n4) [inline muoxi/src/staging/prelude.rs]: This change works

```bash
server_1     | TCP Client listening on 0.0.0.0:8000 proxying to 0.0.0.0:4567
server_1     | New user! on 172.19.0.1:55802
server_1     | thread 'main' panicked at 'Couldn't establish connection to caching server: Connection refused (os error 111)', db/src/cache_structures/socket.rs:48:33
server_1     | note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
muoxi_server_1 exited with code 101
```

But I need to debug this.

Comment (n0um3n4) [reply muoxi/src/staging/prelude.rs]: Probably use a _ENV VAR_

File: muoxi/src/staging/prelude.rs [hunk 2]
```diff
@@ -40,10 +40,10 @@ pub type LinesCodecResult<T> = Result<T, LinesCodecError>;
...
 
 /// Current listening port of the staging proxy server
-pub static PROXY_ADDR: &'static str = "127.0.0.1:8000";
+pub static PROXY_ADDR: &'static str = "0.0.0.0:8000";
```
Comment (n0um3n4) [inline muoxi/src/staging/prelude.rs]: Same as above.

File: dev-entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+lock_setup() { mkdir -p $APP_TEMP_PATH && touch $APP_SETUP_LOCK; }
+unlock_setup() { rm -rf $APP_SETUP_LOCK; }
+wait_setup() { echo "Waiting for app setup to finish..."; sleep $APP_SETUP_WAIT; }
+
+trap unlock_setup HUP INT QUIT KILL TERM EXIT
+
+if [ -z "$1" ]; then set -- cargo run --bin muoxi_staging "$@"; fi
+
+if [ "$1" = "cargo run" ]
```
Comment (n0um3n4) [inline dev-entrypoint.sh:18]: This needs to like in the dockerfile, namely:

```bash
$ cargo install --path=/usr/src/muoxi 
```

**Comment by n0um3n4** [comment]
@duysqubix  I'm getting this in the browser. I think it's working now ?
In the meantime, what do you think it's the best approach to handle the address for the services such as `redis` and `postgres` ? `ENV`?

Browser output

**Comment by duysqubix** [comment]
@thelastinuit I never worked on websocket support yet, so you are in new territory for using browser. I’ve been using MushClient for testing. 

I am not sure what you mean by handling the addresses for the services. Keeping them at default?

**Comment by n0um3n4** [comment]
> @thelastinuit I never worked on websocket support yet, so you are in new territory for using browser. I’ve been using MushClient for testing. 
> 
> 
> 
> I am not sure what you mean by handling the addresses for the services. Keeping them at default?

@duysqubix Let me test with that then.

About the addresses: you can see that to make it work i made some changes from 127.0.0.1 to 0.0.0.0.
So my question is should i use an env var instead and default it to 127.0.0.1 when it's not defined? Do you have something else in mind?

**Comment by n0um3n4** [comment]
@duysqubix I made some changes. Please let me know what you think.

**Comment by duysqubix** [comment]
> > @thelastinuit I never worked on websocket support yet, so you are in new territory for using browser. I’ve been using MushClient for testing.
> > I am not sure what you mean by handling the addresses for the services. Keeping them at default?
> 
> @duysqubix Let me test with that then.
> 
> About the addresses: you can see that to make it work i made some changes from 127.0.0.1 to 0.0.0.0.
> So my question is should i use an env var instead and default it to 127.0.0.1 when it's not defined? Do you have something else in mind?

I do not have anything else in mind at the moment.
Reviewed changes, it is great that you added docker support!

**Pull request merged**